### PR TITLE
MODORDSTOR-79 Remove associated piece records when removing POLine

### PR DESCRIPTION
**Pull request by grabsefx**
## Purpose
* https://issues.folio.org/browse/MODORDSTOR-79

## Approach
* Implemented described logic
* Unit tests updated

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/main/java/org/folio/rest/impl/PoLinesAPI.java [hunk 1]
```diff
@@ -1,13 +1,15 @@
 package org.folio.rest.impl;
 
-import static org.folio.rest.persist.HelperUtils.getEntitiesCollection;
```
Comment (piotr-kalashuk) [inline src/main/java/org/folio/rest/impl/PoLinesAPI.java]: Please keep eclipse style formatting

File: src/main/java/org/folio/rest/impl/PoLinesAPI.java [hunk 2]
```diff
@@ -40,28 +55,140 @@ public void getOrdersStoragePoLines(String query, int offset, int limit, String
...
+    });
+    return future;
+  }
+
+  private CompletableFuture<TxWithId> deletePOLineById(TxWithId tx) {
+    CompletableFuture<TxWithId> future = new CompletableFuture<>();
+
+    String query = String.format(DELETE_POLINE_BY_ID_QUERY, tx.getId());
+    log.info("Delete POLine with id={}", tx.getId());
+    pgClient.execute(tx.getConnection(), query, reply -> {
```
Comment (piotr-kalashuk) [inline src/main/java/org/folio/rest/impl/PoLinesAPI.java]: You can use 
```java
delete(AsyncResult<SQLConnection> conn, String table, Criterion filter, Handler<AsyncResult<UpdateResult>> replyHandler)
```
and build
```java
Criteria criteria = new Criteria();
criteria.addField("'id'").setOperation("=").setValue(id);
```

File: src/main/java/org/folio/rest/impl/PoLinesAPI.java [hunk 3]
```diff
@@ -40,28 +55,140 @@ public void getOrdersStoragePoLines(String query, int offset, int limit, String
...
+        startTxWithId(tx)
+          .thenCompose(this::deletePiecesByPOLineId)
+          .thenCompose(this::deletePOLineById)
+          .thenCompose(this::endTxWithId)
+          .thenAccept(result -> {
+            log.info("Preparing response to client");
+            asyncResultHandler.handle(Future.succeededFuture(DeleteOrdersStoragePoLinesByIdResponse.respond204()));
+          })
+          .exceptionally(t -> {
+            endTxWithId(tx)
```
Comment (piotr-kalashuk) [inline src/main/java/org/folio/rest/impl/PoLinesAPI.java]: Looking at [PostgresClient.rollbackTx(...)](https://github.com/folio-org/raml-module-builder/blob/v24.0.0/domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java#L628) I see that after rollback the connection should be closed.

File: src/main/java/org/folio/rest/impl/PoLinesAPI.java [hunk 4]
```diff
@@ -17,19 +19,32 @@
...
+import javax.ws.rs.core.Response;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+
+import static org.folio.rest.persist.HelperUtils.getEntitiesCollection;
+
 public class PoLinesAPI implements OrdersStoragePoLines {
+  private static final Logger log = LoggerFactory.getLogger(OrdersStoragePoLines.class);
+
+  private static final String PIECES_TABLE = "pieces";
```
Comment (piotr-kalashuk) [inline src/main/java/org/folio/rest/impl/PoLinesAPI.java]: Please use `PiecesAPI.PIECES_TABLE`

File: src/main/java/org/folio/rest/impl/PoLinesAPI.java (hunk 1 above)
Comment (grabsefx) [reply src/main/java/org/folio/rest/impl/PoLinesAPI.java]: eclipse formatter used

File: src/main/java/org/folio/rest/impl/PoLinesAPI.java (hunk 3 above)
Comment (grabsefx) [reply src/main/java/org/folio/rest/impl/PoLinesAPI.java]: fixed rollback transaction

File: src/main/java/org/folio/rest/impl/PoLinesAPI.java (hunk 4 above)
Comment (grabsefx) [reply src/main/java/org/folio/rest/impl/PoLinesAPI.java]: fixed

File: src/main/java/org/folio/rest/impl/PoLinesAPI.java (hunk 2 above)
Comment (grabsefx) [reply src/main/java/org/folio/rest/impl/PoLinesAPI.java]: updated to use `delete` method

**Pull request merged**